src/app/shared-module/services/task.service.ts

```diff
@@ -120,100 +199,34 @@ export class TaskService {
     );
   }
 
-  private getTasks(): TaskForList[] {
-    return [
-      {
-        id: '1',
-        title: 'task1',
-        frequency: 'Day',
-        allocatedTime: 1000,
-        progress: 50,
-      },
-      {
-        id: '2',
-        title: 'task2',
-        frequency: 'Week',
-        allocatedTime: 600,
-        progress: 10,
-      },
-      {
-        id: '3',
-        title: 'task3',
-        frequency: 'Month',
-        allocatedTime: 30,
-        progress: 25,
-      },
-      {
-        id: '4',
-        title: 'task4',
-        frequency: 'Day',
-        allocatedTime: 1500,
-        progress: 100,
-      },
-      {
-        id: '5',
-        title: 'task5',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 60,
-      },
-      {
-        id: '6',
-        title: 'task6',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 100,
-      },
-      {
-        id: '7',
-        title: 'task7',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 35,
-      },
-      {
-        id: '8',
-        title: 'task8',
-        frequency: 'Day',
-        allocatedTime: 1600,
-        progress: 40,
-      },
-      {
-        id: '9',
-        title: 'task9',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 55,
-      },
-      {
-        id: '10',
-        title: 'task10',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 80,
-      },
-      {
-        id: '11',
-        title: 'task11',
-        frequency: 'Day',
-        allocatedTime: 1100,
-        progress: 95,
-      },
-    ];
+  private isTaskForToday(task: Task): boolean {
+    const taskDate = new Date(task.initialDate);
+    const today = new Date();
+    if (
+      taskDate.getFullYear() >= today.getFullYear() &&
+      taskDate.getMonth() >= today.getMonth() &&
+      taskDate.getDate() > today.getDate()
+    ) {
+      return false;
+    }
+    if (task.frequency.frequencyValue === TaskFrequenciesEnum.Weekend) {
+      return today.getDay() == 6 || today.getDay() == 0;
+    }
+    if (task.frequency.frequencyValue === TaskFrequenciesEnum.Workday) {
+      return today.getDay() != 6 && today.getDay() != 0;
+    }
+    return true;
   }
 
-  private getTask(): Task {
+  private createPomodoro(): Pomodoro {
     return {
-      id: '11',
-      title: 'task11',
-      frequency: {
-        id: '145',
-        frequencyType: TaskFrequenciesEnum.Day,
-        every: 1,
-        isCustom: false,
-      },
-      allocatedTime: 1100,
-      initialDate: new Date(2023, 1, 20),
+      id: '00000000-0000-0000-0000-000000000000',
```

Comment:
Just a small remark: you can use `empty()` method located in the `src\app\shared-module\types\guid.ts` for this.

Reply:
done